### ISSUE TITLE
IDE Crash while editing file

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#2045]
Code to reproduce:
```
using System;

// type here: 

namespace System
{
	extension Array
	{
		public static int BinarySearch<T, TAlt, ComparerFN>(Span<T> data, TAlt value, ComparerFN comp)
			where ComparerFN : delegate int (T lhs, TAlt rhs)
		{
			int lo = 0;
			int hi = data.Length - 1;

			while (lo <= hi)
			{
			    int i = (lo + hi) / 2;
				T midVal = data[i];
				int c = comp(midVal, value);
			    if (c == 0) return i;
			    if (c < 0)
			        lo = i + 1;                    
			    else
			        hi = i - 1;
			}
			return ~lo;
		}

		public static int BinarySearch<T, TAlt>(Span<T> data, TAlt value)
			where int : operator T <=> TAlt
		{
			int lo = 0;
			int hi = data.Length - 1;

			while (lo <= hi)
			{
			    int i = (lo + hi) / 2;
				T midVal = data[i];
				int c = midVal <=> value;
			    if (c == 0) return i;
			    if (c < 0)
			        lo = i + 1;                    
			    else
			        hi = i - 1;
			}
			return ~lo;
		}	
	}
}
```
Editing the file at the top will make IDE crash without showing crash handler.

Tested with: Beef IDE Version 0.43.5 (Nightly 10/19/2024) Build 691c147089d9abcef97f327a489fdadf2331e018

**Comment by bfiete** [comment]
Can't reproduce. I think something else is required. Can you post a repro program?

**Comment by Fusioon** [comment]
[TestEmpty.zip](https://github.com/user-attachments/files/17497456/TestEmpty.zip)
Sorry for not checking properly in empty project, here is a project where i can consistently trigger this, open `Utils.bf` and type something at the top, you should see IDE just close.

Also same behavior can be observed by doing the same in `ipfs/Multibase.bf`

**Comment by bfiete** [comment]
I found issues but not related to this array extension, it was related to the code emissions in `Multibase.EBase`...

Fixed at 35271626aa2e13f2dbbe35727046e18d0b4ad1b2

**Comment by Fusioon** [comment]
Well that type was also in my test project when i created the issue, so it makes sense. Thanks for the fix
